src/cargo/util/toml.rs

```diff
@@ -426,28 +437,46 @@ impl TomlManifest {
         let blacklist = vec!["build", "deps", "examples", "native"];
 
         for bin in bins.iter() {
-            if blacklist.iter().find(|&x| *x == bin.name) != None {
+            if blacklist.iter().find(|&x| *x == bin.name.as_ref().unwrap()) != None {
```

Comment:
I think that this `unwrap` can trigger a panic if the `name` is omitted on a `[[bin]]` section.